### DISCUSSION TITLE
Tangelo install for Mac users (including Dockerfile)

### DISCUSSION THREAD

**Discussion by ValentinS4t1qbit** [goodchemistryco/Tangelo#190] [General]
Installation of the package on Mac is really frustrating because a number of dependencies are not easily building with Apple C/C++ compiler clang (who's masquerading as gcc/g++). I don't want people working on MacOS to feel excluded, and in general I don't want any poor soul to waste their time on installation, when more interesting science could be done instead.

I think we should:

1. Update / fix the Dockerfile, which hasn't been touched in a while. Maybe refactor setup.py and Dockerfile to make sure they always have similar install instructions. This should let Docker-savvy users deploy an ubuntu environment that "just works" from whatever OS they are using. We should probably link some resources to then help people use the docker image.

2. Try to make a guide for people who want to set up everything in their MacOS environment.
Like, installing GNU gcc/g++ with brew, creating aliases for the compilers, linking openblas and lapack, building pyscf and its dependencies in any way that works and is compatible with Tangelo setup...

I wonder what would resonate most with our Mac users here (and how many there are), in particular. Any comments, or suggestion ?

**Comment by ValentinS4t1qbit** [comment]
Update: 

- the docker has been fixed in August
- notebooks are easily run under Google Collab, which provides a "no local install" path for users who need to spin up an environment quickly to try something.
- @AlexandreF-1qbit is looking into a "no pyscf" install in order to help people who do not need chemistry functionalities for their project, or would handle pyscf installation separately.

**Comment by alexfleury-sb** [comment]
Maybe a [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml//) ([PEP 518](https://peps.python.org/pep-0518/), [PEP 517](https://peps.python.org/pep-0517/), [PEP 621](https://peps.python.org/pep-0621/) and [PEP 660](https://peps.python.org/pep-0660/)) file would be handy in our situation. It is supposed to isolate the build environment, therefore it might fix the errors we have been experiencing while building `pyscf` on MacOS. I believe `setup.py` and `setup.cfg` will be deprecated in the future, and they will be replaced by a pyproject.toml file.

I did some experiments with the tangelo package and it works remarkably, **EXCEPT**:
- In the situation where one has an older version of `setuptools` (`setuptools<59.0` doesn't recognize the package structure and it installs a `UNKNOWN-0.0.0` package). This is the case on GColab.
- However, there is also a bug in pip where the isolation is broken (therefore defeating the purpose of the file). A fix has been merged recently to address the building environment isolation (https://github.com/pypa/pip/pull/11466).

**tl;dr**: On MacOS 11.6.7 (intel) and Ubuntu 20.04.5 LTS, installing from a pyproject.toml file works. On GColab, it doesn't work.
